### ISSUE TITLE
Add top-N logprobs alongside token generation

### ISSUE THREAD

**Issue by sauravpanda** [sauravpanda/flarellm#228]
## Summary

After each token is generated, expose the top-N (token_id, log_probability) pairs alongside the sampled token. This enables per-token confidence scores, classification, and re-ranking in the browser.

## Motivation

- Sequence classification using token probabilities
- Visualizing model confidence during streaming generation
- Token-level log-probability for perplexity calculation

## API

```typescript
// Set how many top logprobs to track (0 = disabled)
set_top_logprobs(n: number): void

// After each next_token() call, returns Float32Array of [token_id_0, logprob_0, token_id_1, logprob_1, ...]
get top_logprobs(): Float32Array
```

## Implementation Notes

- `last_logits` (added in #225) already captures the raw logit vector
- Apply softmax + log to get log-probabilities, then take the top-N by descending logprob
- Store result as interleaved `Vec<f32>` with (token_id as f32, logprob) pairs — avoids needing a custom JS-visible struct
- Capture after forward pass, before sampling (same point as `last_logits`)
